src/audio-worklet/free-queue/src/interface/C/free_queue.h

```diff
@@ -0,0 +1,206 @@
+#ifndef FREE_QUEUE_C
+#define FREE_QUEUE_C
+
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdatomic.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef EMSCRIPTEN_KEEPALIVE
+#define EMSCRIPTEN_KEEPALIVE __attribute__((used))
+#endif
+
+/**
+ * C Struct to represent FreeQueue
+*/
+struct FreeQueue {
+    size_t bufferLength;
+    size_t channelCount;
+    atomic_uint* state;
+    float** channels;    
+};
+
+enum FQState {
+    READ = 0,
+    WRITE = 1
+};
+
+/**
+ * C API for implementing and acessing FreeQueue.
+ */
+
+/**
+ * Create a FreeQueue and returns pointer.
+ * Takes length of FreeQueue and channel Count as parameters.
+ * Returns pointer to created FreeQueue.
+*/
+EMSCRIPTEN_KEEPALIVE struct FreeQueue* create_free_queue(size_t length, size_t channelCount);
+
+/**
+ * Push new data to FreeQueue.
+ * Takes pointer to FreeQueue, pointer to input data, and block length as parameters.
+ * Returns if operation was successful or not as boolean.
+*/
+EMSCRIPTEN_KEEPALIVE bool free_queue_push(struct FreeQueue* fq, float** input, size_t blockLength);
+
+/**
+ * Pull data from FreeQueue.
+ * Takes pointer to FreeQueue, pointer to output buffers, and block length as parameters.
+ * Returns if operation was successful or not as boolean.
+*/
+EMSCRIPTEN_KEEPALIVE bool free_queue_pull(struct FreeQueue* fq, float** output, size_t blockLength);
+
+/**
+ * Destroy FreeQueue.
+ * Takes pointer to FreeQueue as parameter.
+*/
+EMSCRIPTEN_KEEPALIVE void destroy_free_queue(struct FreeQueue* fq);
+
+/**
+ * Helper Function to get Pointers to data members of FreeQueue Struct.
+ * Takes pointer to FreeQueue, and char* string refering to data member to query.
+*/
+EMSCRIPTEN_KEEPALIVE void* get_free_queue_pointers(struct FreeQueue* fq, char* data);
+
+#ifdef FREE_QUEUE_IMPL
+
+static uint32_t _getAvailableRead(struct FreeQueue* fq, uint32_t readIndex, uint32_t writeIndex) {
```

Comment:
Add an overview comment for the function. Also why do we have an underscore prefix?

Reply:
I saw this underscore naming convention being used in JavaScript code for private functions, so I thought of doing same, that's why I made function static too